src/WebJobs.Script/Description/Node/Compilation/JavaScriptCompilationServiceFactory.cs

```diff
@@ -21,11 +33,40 @@ public ICompilationService<IJavaScriptCompilation> CreateService(ScriptType scri
                 case ScriptType.Javascript:
                     return new RawJavaScriptCompilationService(metadata);
                 case ScriptType.TypeScript:
-                    return new TypeScriptCompilationService();
+                    return new TypeScriptCompilationService(_typeScriptOptions.Value);
                 default:
                     throw new NotSupportedException(string.Format(CultureInfo.InvariantCulture,
                         "The script type {0} is not supported by the {1}", scriptType, typeof(JavaScriptCompilationServiceFactory).Name));
             }
         }
+
+        private TypeScriptCompilationOptions CreateTypeScriptCompilationOptions()
+        {
+            return new TypeScriptCompilationOptions
+            {
+                ToolPath = GetTypeScriptToolPath(),
+                OutDir = ".output",
+                RootDir = _host.ScriptConfig.RootScriptPath
+            };
+        }
+
+        private static string GetTypeScriptToolPath()
+        {
+            string path = ScriptSettingsManager.Instance.GetSetting(EnvironmentSettingNames.TypeScriptCompilerPath);
+            if (path == null)
+            {
+                string basePath = Path.Combine(Environment.ExpandEnvironmentVariables("%programfiles(x86)%"), "Microsoft SDKs\\TypeScript");
+                if (Directory.Exists(basePath))
+                {
+                    path = Directory.GetDirectories(basePath)
+                        .OrderByDescending(d => d)
+                        .FirstOrDefault() ?? string.Empty;
```

Comment:
Yes, that is the current behavior. This is currently by design, but likely to change (we'll likely be packaging TS as part of the runtime, but there are some other ideas).
Kept it this way here as it is outside of the scope of this changes, but something we'll be addressing in the future.